pkg/txn/storage/memorystorage/memorytable/types.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memorytable
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/vm/engine/memoryengine"
+)
+
+type ID = memoryengine.ID
+
+type Text string
+
+func (t Text) Less(than Text) bool {
+	return t < than
+}
+
+type Bool bool
+
+func (b Bool) Less(than Bool) bool {
+	return bool(!b && than)
+}
+
+type Int int64
+
+func (i Int) Less(than Int) bool {
+	return i < than
+}
+
+type Uint int64
+
+func (i Uint) Less(than Uint) bool {
+	return i < than
+}
+
+type Float float64
+
+func (f Float) Less(than Float) bool {
+	return f < than
+}
+
+type Bytes []byte
+
+func (b Bytes) Less(than Bytes) bool {
+	return bytes.Compare(b, than) < 0
+}
+
+func ToOrdered(v any) any {
```

Comment:
This file should be in container/types.    Reuse as much as possible, if cannot, make new code as reusable as possible.

Reply:
I suggest the first step is to add Int, Bool, etc to container/types and replace the existing int, bool, etc. This will affect many codes.